### PR TITLE
Docs updates for upcoming 8.12.0/9.4.0 release of Umbraco Forms.

### PR DESCRIPTION
**Pull request by AndyButland**
I'm not sure exactly when this will be needed yet, but earliest would be 5th April for the release candidate - which is when we should update the docs.

### CODE REVIEW

File: Add-ons/UmbracoForms/Developer/Custom-Markup/index.md [hunk 1]
```diff
@@ -65,6 +65,16 @@ Umbraco Forms uses ASP.NET Unobtrusive Validation which is why you see attribute
 
 This can be customized but it's important to keep the ID of the control to `@Model.Id` since that is used to match the value to the Form field.  For fields that are conditionally hidden, without an ID of `@Model.Id` the control won't be shown when the conditions to show the field are met.  An ID needs to be added to text controls such as headings and paragraphs.
 
+The view model for the partial view for a field is `FieldViewModel`. This defined various properties that may be useful when creating new themes or custom fields, some of which are shown in the code samples above.  Others include:
```
Comment (eshanrnh) [inline Add-ons/UmbracoForms/Developer/Custom-Markup/index.md:68]: The view model for the partial view of a field is `FieldViewModel`. This defines various properties that may be useful when creating new themes or custom fields, some of which are shown in the code samples above.  Others include:

Comment (eshanrnh) [inline Add-ons/UmbracoForms/Developer/Custom-Markup/index.md:68]: The view model for the partial view of a field is `FieldViewModel`. This defines various properties that may be useful when creating new themes or custom fields, some of which are shown in the code samples above.  Others include:

File: Add-ons/UmbracoForms/Developer/Custom-Markup/index.md [hunk 2]
```diff
@@ -65,6 +65,16 @@ Umbraco Forms uses ASP.NET Unobtrusive Validation which is why you see attribute
...
+
+- `AdditionalSettings` - a dictionary of the settings keys and values populated for the form field. These can be retrieved in typed form by key using e.g. `Model.GetSettingValue<int>("MaximumLength", 255);`.
+
+The following are avaialble on the model but only populated for fields that support file upload:
```
Comment (eshanrnh) [inline Add-ons/UmbracoForms/Developer/Custom-Markup/index.md:72]: The following are avaialble on the model but only populated for fields that support File Upload:

Comment (eshanrnh) [inline Add-ons/UmbracoForms/Developer/Custom-Markup/index.md:72]: The following are avaialble on the model but only populated for fields that support File Upload:

File: Add-ons/UmbracoForms/Developer/Security/index.md [hunk 3]
```diff
@@ -51,10 +51,17 @@ With that in place the _Form Security_ tree divides into three sub-trees:
 
 - Under _User Permissions_, each user that has a specific user permission record is listed and can be managed.  Records for users can be created or deleted via the tree's action menu.
 
-- As we no longer necessarily have a user permission record for every user in the system, _Start folders_ are moved into a separate area where all users are listed and hence the appropriate options can be set.
-
 As new forms are created, user groups with aliases listed in the `GrantAccessToNewFormsForUserGroups` configuration setting will be automatically given access.  For example, with a value of `admin,editor`, the built-in Adminstrators and Editors groups would have access.
 
+### Start folders for user groups
+
+In Umbraco Forms 8.12.0 and 9.4.0, start folders were also enabled for user groups.  They work in a similar way to the group based permissions described above:
```
Comment (eshanrnh) [inline Add-ons/UmbracoForms/Developer/Security/index.md:58]: In Umbraco Forms 8.12.0 and 9.4.0, start folders are also enabled for User Groups.  They work in a similar way as the group based permissions described above:

Comment (eshanrnh) [inline Add-ons/UmbracoForms/Developer/Security/index.md:58]: In Umbraco Forms 8.12.0 and 9.4.0, start folders are also enabled for User Groups.  They work in a similar way as the group based permissions described above:

File: Add-ons/UmbracoForms/Developer/Security/index.md [hunk 4]
```diff
@@ -51,10 +51,17 @@ With that in place the _Form Security_ tree divides into three sub-trees:
...
+
+- If the user has a specific user permission set, it is used in preference to anything set on the user groups they are a part of.
+    - This means if the user has no start folders defined, even if they groups they are part of do, they will have access to the root of the forms tree and be able to access all folders and forms.
```
Comment (eshanrnh) [inline Add-ons/UmbracoForms/Developer/Security/index.md:61]: This means if the user has no start folders defined and the groups they are part of do, they will have access to the root of the Forms tree and be able to access all folders and Forms.

Comment (eshanrnh) [inline Add-ons/UmbracoForms/Developer/Security/index.md:61]: This means if the user has no start folders defined and the groups they are part of do, they will have access to the root of the Forms tree and be able to access all folders and Forms.

File: Add-ons/UmbracoForms/Developer/Security/index.md [hunk 5]
```diff
@@ -51,10 +51,17 @@ With that in place the _Form Security_ tree divides into three sub-trees:
...
+ - If the user doesn't have a specific user permission set, they are granted access to all the unique folders the groups they are part of have access to.
+    - If they are part of any group that has no start folders defined, they will have access to the root of the forms tree and be able to access all folders and forms.
```
Comment (eshanrnh) [inline Add-ons/UmbracoForms/Developer/Security/index.md:63]: If they are part of any group that has no start folders defined, they will have access to the root of the Forms tree and be able to access all folders and Forms.

Comment (eshanrnh) [inline Add-ons/UmbracoForms/Developer/Security/index.md:63]: If they are part of any group that has no start folders defined, they will have access to the root of the Forms tree and be able to access all folders and Forms.

**Review by eshanrnh — COMMENTED** [review]
@AndyButland Looks good. Just couple of suggestions.

File: Add-ons/UmbracoForms/Developer/Configuration/index-v7.md [hunk 6]
```diff
@@ -86,6 +86,11 @@ Added in 8.7.0, this setting controls the maximum number of columns that can be
 ### DefaultTheme
 Added in 8.8.0, this setting allows you to configure the name of the theme to use when an editor has not specifically selected one for a form.  If empty or missing, the default value of "default" is used.  If a custom default theme is configured, it will be used for rendering forms where the requested file exists, and where not, will fall back to the out of the box default theme.
 
+### DefaultEmailTemplate
+When creating an empty form, a single workflow is added that will send an email to the current user's address. By default, the template shipped with Umbraco Forms and found at `Forms/Emails/Example-Template.cshtml` is used.
```
[the remark this answers is not in the document]
Comment (AndyButland) [reply Add-ons/UmbracoForms/Developer/Configuration/index-v7.md:90]: Thanks - will update.

File: Add-ons/UmbracoForms/Developer/Configuration/index-v7.md [hunk 7]
```diff
@@ -86,6 +86,11 @@ Added in 8.7.0, this setting controls the maximum number of columns that can be
...
+
+If you have created a custom template and would like to use that as the default instead, you can set the path here using this configuration setting.
```
[the remark this answers is not in the document]
Comment (AndyButland) [reply Add-ons/UmbracoForms/Developer/Configuration/index-v7.md:92]: I think this is OK as is (and consistent with all the other configuration keys discussed on this page).  For each of these configuration settings the title of the section - i.e. "DefaultEmailTemplate" in this case - is the configuration key to use.

[the remark this answers is not in the document]
Comment (AndyButland) [reply Add-ons/UmbracoForms/Developer/Configuration/index-v7.md:92]: As above.

**Comment by AndyButland** [comment]
Thanks @eshanrnh, I've updated based on your suggestions other than the one point I've commented on.

**Comment by eshanrnh** [comment]
> Thanks @eshanrnh, I've updated based on your suggestions other than the one point I've commented on.

That's a valid point and perfectly fine 😊

**Review by eshanrnh — APPROVED** [review]

**Comment by jonat123** [comment]
Needs to be merged in on tuesday the 12th of april

**Pull request merged**